### PR TITLE
Congrats mail cron

### PR DESCRIPTION
**Pull request by mnmallea**
## Summary
- Added cron to send congratulation mails.
- Mails are sent to top 10 users with more words weeted since the last day.
- Runs every day at 0:00. Por qué? No hay por qué.
- Added specs.

## Known Issues
None.

## Screenshots
image

## Trello Card
https://trello.com/c/SPsRNlLI/39-proceso-de-env%C3%ADo-de-mail-de-felicitaci%C3%B3n

### CODE REVIEW

File: app/services/mailer.js [hunk 1]
```diff
@@ -24,16 +24,32 @@ const getTransporter = () => {
...
-      to: user.email,
-      subject: 'Bienvendio a Weeter!',
-      text: `Hola ${user.firstName}. Tu cuenta ha sido creada correctamente en Weeter`
-    })
-    .catch(({ message }) => {
-      throw externalServiceError(`Error when sending welcome email to user ${user.id}. Error: ${message}`);
-    });
+  return sendMailToUser(user, {
+    subject: 'Bienvendio a Weeter!',
+    text: `Hola ${user.firstName}. Tu cuenta ha sido creada correctamente en Weeter`
```
Comment (lnmedrano) [inline app/services/mailer.js:39]: Messages are usually set in env vars. That way, if we see a typo ('bienvendio'), or we want to change the message, we can change the env and re-deploy, without having to go through the entire pr proccess.

File: app/services/mailer.js [hunk 2]
```diff
@@ -24,16 +24,32 @@ const getTransporter = () => {
...
+  });
+};
+
+exports.sendCongratulatoryEmail = (user, wordsCount) => {
+  info(`mailer.sendCongratulatoryEmail to user ${user.id}`);
+  return sendMailToUser(user, {
+    subject: 'Felicitaciones por Weetear!',
+    text: `Hola ${user.firstName}. Tus weets están llegando a mucha gente.
+    Hoy weeteaste ${wordsCount} palabras y ya acumulaste ${user.points} puntos!!!
+    `
```
Comment (lnmedrano) [inline app/services/mailer.js:51]: Idem with messages

File: app/services/users.js [hunk 3]
```diff
@@ -16,6 +21,13 @@ exports.findUserByEmail = email => {
   });
 };
 
+exports.findAllUsers = ({ offset, limit }) => {
+  info(`Calling users.findAllUsers offset=${offset}, limit=${limit}`);
+  return User.findAll({ offset, limit, order: [['id', 'asc']] }).catch(error => {
+    throw databaseError(error.message);
+  });
+};
+
```
Comment (lnmedrano) [inline app/services/users.js:30]: Is this used?

File: app/services/users.js [hunk 4]
```diff
@@ -44,6 +56,36 @@ exports.modifyUserPointsBy = (user, by, transaction) => {
...
+          'sum',
+          sequelize.fn(
+            'array_length',
+            sequelize.fn('regexp_split_to_array', sequelize.col('content'), sequelize.literal("E'\\\\s+'")),
+            1
+          )
+        ),
+        'totalWordsCount'
+      ]
+    ],
```
Comment (lnmedrano) [inline app/services/users.js:79]: Nice :sunglasses:

**Review by lnmedrano — CHANGES_REQUESTED** [review]

File: app/services/users.js (hunk 3 above)
Comment (mnmallea) [reply app/services/users.js:30]: no, f

**Review by lnmedrano — APPROVED** [review]

**Pull request merged**